fluss-lake/fluss-lake-iceberg/src/main/java/com/alibaba/fluss/lake/iceberg/IcebergLakeCatalog.java

```diff
@@ -204,21 +208,47 @@ private PartitionSpec createPartitionSpec(
                                         new IllegalArgumentException(
                                                 "Bucket count (bucket.num) must be set"));
 
-        if (bucketKeys.isEmpty()) {
-            throw new IllegalArgumentException(
-                    "Bucket key must be set for primary key Iceberg tables");
-        }
+        // Only support one bucket key for now
         if (bucketKeys.size() > 1) {
             throw new UnsupportedOperationException(
                     "Only one bucket key is supported for Iceberg at the moment");
         }
 
+        // pk table must have bucket key
+        if (bucketKeys.isEmpty() && isPkTable) {
+            throw new IllegalArgumentException(
+                    "Bucket key must be set for primary key Iceberg tables");
+        }
+
+        // bucket key must exist in schema
```

Comment:
We can remove this check since it already been checked before call method `IcebergLakeCatalog#createTable`